model/src/main/java/org/projectnessie/model/Merge.java

```diff
@@ -30,16 +30,23 @@
     type = SchemaType.OBJECT,
     title = "Merge Operation",
     // Smallrye does neither support JsonFormat nor javax.validation.constraints.Pattern :(
-    properties = {@SchemaProperty(name = "fromHash", pattern = Validation.HASH_REGEX)})
+    properties = {
+      @SchemaProperty(name = "sourceRefName", pattern = Validation.REF_NAME_REGEX),
+      @SchemaProperty(name = "fromHash", pattern = Validation.HASH_REGEX)
+    })
 @Value.Immutable(prehash = true)
 @JsonSerialize(as = ImmutableMerge.class)
 @JsonDeserialize(as = ImmutableMerge.class)
 public interface Merge {
 
-  @NotBlank
+  @Nullable
```

Comment:
Can you explain the changes to Transplant and Merge a little bit? It looks to me like `Merge` is now actually a reference?

Reply:
Huh? Where's that change?

Reply:
k, the annotation makes more sense now. Still can you explain the changes to Transplant and Merge a little bit? It looks to me like Merge is now actually a reference?

Reply:
Sorry, still don't understand _"It looks to me like Merge is now actually a reference?"_
It's only that merge+transplant not just take a hash but also where the hash is coming from.

Reply:
and a `Reference` is a a hash and a name. So we are saying a Merge is moving commits from ReferenceA to ReferenceB

Reply:
You mean it _should_ be a `Reference`?

Reply:
I mean it could be a reference, I am asking if we should think about how we want `Merge` and `Transplant` to look long term and what makes sense as a model

Reply:
Ah - yea, it should have those semantics via `org.projectnessie.client.api.NessieApiV1#mergeRefIntoBranch`. Will add that there. Tricky for transplant (cherry-pick) as there are many hashes.